### DISCUSSION TITLE
parking_model_based #removing the dependancy.

### DISCUSSION THREAD

**Discussion by ss555** [eleurent/highway-env#236] [Q&A]
I was studying 
[parking_model_based](https://colab.research.google.com/github/eleurent/highway-env/blob/master/scripts/parking_model_based.ipynb#scrollTo=F7Gl2kKJfviu)
DynamicsModel and wondered the use of ??
xu[:, self.STATE_X:self.STATE_Y+1] = 0  # Remove dependency in (x,y)
I work with another envoronment and it seems to work without removing the dependancy.
the line before `xu = torch.cat((x, u), -1)` just the concatenates state with the control

**Comment by eleurent** [comment]
Hi,
This line is just here to help training faster by encoding the knowledge (it is an inductive bias) that the dynamics (state derivative), which normally depend on the full state (x, y, angle, velocity), actually only depend on the angle and velocity here, and are the same in every position in space. 
Note that this is not a general law, you can imagine plenty of scenario where this is not true (eg the friction coefficients of the ground could change, etc). 
And note that this is not required as well, as the training procedure could learn that the coefficients associated with x and y are always 0. It should just take a little bit longer to train, and may not generalize to states that were not seen during training. 

Does that make sense?
